app/uk/gov/hmrc/tai/service/TrackingService.scala

```diff
@@ -30,26 +30,62 @@ import uk.gov.hmrc.tai.util.constants.JourneyCacheConstants
 
 import scala.util.control.NonFatal
 
-class TrackingService @Inject() (trackingConnector: TrackingConnector,
-                                 @Named("Track Successful Journey") successfulJourneyCacheService: JourneyCacheService
-                                ) extends JourneyCacheConstants{
+sealed trait TimeToProcess
 
-  def isAnyIFormInProgress(nino: String)(implicit hc: HeaderCarrier): Future[Boolean] = {
-    val trackingStatus = trackingForTesForms(nino) map { trackedForm =>
-      !trackedForm.forall(_.status == TrackedFormDone)
-    } recover {
-      case NonFatal(x) => Logger.warn(s"Tracking service returned error, therefore return false in response. Error: ${x.getMessage}")
-        false
-    }
+case object ThreeWeeks extends TimeToProcess
+
+case object SevenDays extends TimeToProcess
+
+case object NoTimeToProcess extends TimeToProcess
+
+class TrackingService @Inject()(trackingConnector: TrackingConnector,
+                                @Named("Track Successful Journey") successfulJourneyCacheService: JourneyCacheService
+                               ) extends JourneyCacheConstants {
+
+  def isAnyIFormInProgress(nino: String)(implicit hc: HeaderCarrier): Future[TimeToProcess] = {
+
+    val longTESProcesses: Future[Seq[TrackedForm]] = unfinishedTrackingForms(nino, "TES[1|7]")
+    val shortTESProcesses: Future[Seq[TrackedForm]] = unfinishedTrackingForms(nino, "TES[2-6]")
 
     for {
-      isStatusAvailable <- trackingStatus
-      isAnyJourneySuccessful <- successfulJourneyCacheService.currentCache map (_.nonEmpty)
-    } yield isStatusAvailable || isAnyJourneySuccessful
+      haveAnyLongProcesses <- longTESProcesses map (_.nonEmpty)
+      haveAnyShortProcesses <- shortTESProcesses map (_.nonEmpty)
```

Comment:
There is a slight bit of repetition.The following reduces repetition but might be less readable, what do you think?

if(haveAnyShortProcesses || (successfulJournies.nonEmpty && !isA35DayJourney(successfulJournies))){
        SevenDays
      }else if(haveAnyLongProcesses || (successfulJournies.nonEmpty && isA35DayJourney(successfulJournies))){
        ThreeWeeks
      } else {
        NoTimeToProcess
      }

or

(haveAnyShortProcesses, haveAnyLongProcesses, successfulJournies.nonEmpty, isA35DayJourney(successfulJournies)) match {
        case(true, _, _, _) | (_, _, true, false) => SevenDays
        case(_, true, _, _) | (_, _, true, true) => ThreeWeeks
        case(_, _, _, _) =>  NoTimeToProcess
      }